pkg/csi_driver/multishare_ops_manager_test.go

```diff
@@ -573,6 +622,253 @@ func TestListInstanceForStorageClassPrefix(t *testing.T) {
+						Name:            "shared-network",
+						Ip:              "10.0.0.2",
+					},
+					Tier:       "standard",
+					KmsKeyName: "projects/tyuchn-gke-dev/locations/us-central1/keyRings/test-cmek-key-ring/cryptoKeys/test-cmek-key",
+				},
+				{
+					Name:     "test-instance-2",
+					Project:  "test-project",
+					Location: "us-central1",
+					Labels: map[string]string{
+						util.ParamMultishareInstanceScLabelKey: "testprefix",
+					},
+					Network: file.Network{
+						ReservedIpRange: "10.0.0.0/24",
+						ConnectMode:     directPeering,
+						Name:            "default",
+						Ip:              "10.0.0.2",
+					},
+					Tier:       "enterprise",
+					KmsKeyName: "projects/tyuchn-gke-dev/locations/us-central1/keyRings/test-cmek-key-ring/cryptoKeys/test-cmek-key",
+				},
+				{
+					Name:     "test-instance-3",
+					Project:  "test-project",
+					Location: "us-central1",
+					Labels: map[string]string{
+						util.ParamMultishareInstanceScLabelKey: "testprefix",
+					},
+					Network: file.Network{
+						ReservedIpRange: "10.0.0.0/24",
+						ConnectMode:     directPeering,
+						Name:            "shared-network",
+						Ip:              "10.0.0.2",
+					},
+					Tier:       "enterprise",
+					KmsKeyName: "projects/tyuchn-gke-dev/locations/us-central1/keyRings/test-cmek-key-ring/cryptoKeys/test-cmek-key-1",
+				},
+				{
+					Name:     "test-instance-4",
+					Project:  "test-project",
+					Location: "us-central1",
+					Labels: map[string]string{
+						util.ParamMultishareInstanceScLabelKey: "testprefix",
+					},
+					Network: file.Network{
+						ReservedIpRange: "10.0.0.0/24",
+						ConnectMode:     directPeering,
+						Name:            "shared-network",
+						Ip:              "10.0.0.2",
+					},
+					Tier: "enterprise",
+				},
+				{
+					Name:     "test-instance-5",
+					Project:  "test-project",
+					Location: "us-central1",
+					Labels: map[string]string{
+						util.ParamMultishareInstanceScLabelKey: "testprefix",
+					},
+					Network: file.Network{
+						ReservedIpRange: "10.0.0.0/21",
+						ConnectMode:     directPeering,
+						Name:            "shared-network",
+						Ip:              "10.0.0.2",
+					},
+					Tier:       "enterprise",
+					KmsKeyName: "projects/tyuchn-gke-dev/locations/us-central1/keyRings/test-cmek-key-ring/cryptoKeys/test-cmek-key",
+				},
+			},
+			expectedList: []*file.MultishareInstance{
+				{
+					Name:     "test-instance-5",
+					Project:  "test-project",
+					Location: "us-central1",
+					Labels: map[string]string{
+						util.ParamMultishareInstanceScLabelKey: "testprefix",
+					},
+					Network: file.Network{
+						ReservedIpRange: "10.0.0.0/21",
+						ConnectMode:     directPeering,
+						Name:            "shared-network",
+						Ip:              "10.0.0.2",
+					},
+					Tier:       "enterprise",
+					KmsKeyName: "projects/tyuchn-gke-dev/locations/us-central1/keyRings/test-cmek-key-ring/cryptoKeys/test-cmek-key",
+				},
+			},
+		},
```

Comment:
please also add test cases that return error from the function isMatchedInstance()

Reply:
Done. Added a testcase when given CIDR is invalid.